src/tbbbind/tbb_bind.cpp

```diff
@@ -223,6 +223,15 @@ class platform_topology {
         }
     }
 
+    // Explicitly call the interface that was introduced in the HWLOC 2.5 release to prevent linking
+    // HWLOC 2.4 during the tbbbind_2_5 library loading.
+    void backward_compatibility_guarantee() {
```

Comment:
```suggestion
    void enforce_hwloc_2_5_runtime_linkage() {
        // Explicitly call the interface that was introduced in the HWLOC 2.5 release to prevent linking
        // HWLOC 2.4 during the tbbbind_2_5 library loading.
```

Reply:
```suggestion
    void enforce_hwloc_2_5_runtime_linkage() {
        // Without the call of this function HWLOC 2.4 can be successfully loaded during the tbbbind_2_5 loading.
        // It is possible since tbbbind_2_5 don't use any new entry points that were introduced in HWLOC 2.5
        // But tbbbind_2_5 compiles with HWLOC 2.5 header, therefore such situation requires binary forward compatibility
        // which are not guaranteed by the HWLOC library. To enforce linkage tbbbind_2_5 only with HWLOC >= 2.5 version
        // this function calls the interface that is available in the HWLOC 2.5 only.
```